docs/gravity-bridge-setup-guide.md

```diff
@@ -0,0 +1,105 @@
+/tmp/test-cronos` to get a local Cronos testnet.
+
+You should own some funds in this testnet, for the local testnet, you'll get the funds with the same private key as
+above.
+
+## Generate Orchestrator Keys
+
+You need to prepare two accounts for the orchestrator, one for ethereum and one for cronos, you can also use a same
+private key for both accounts. You should transfer some funds to these accounts, so the orchestrator can cover the gas
+fees of messages relaying later.
+
+## Sign Validator Address
+
+To register the orchestrator with the validator, you need to sign a protobuf encoded message using the orchestrator's
+ethereum key, and send it to a cronos validator to register it.
+
+The protobuf message is like this:
+
+```protobuf
+message DelegateKeysSignMsg {
+  // The valoper prefixed cronos validator address
+  string validator_address = 1;
+  // Current nonce of the validator account
+  uint64 nonce = 2;
+}
+```
+
+Use your favorite protobuf library to encode the message, and use your favorite web3 library to do the messge signing,
+for example, this is how it could be done in python:
+
+```python
+msg = DelegateKeysSignMsg(validator_address=val_addr, nonce=nonce)
+sign_bytes = eth_utils.keccak(msg.SerializeToString())
+
+acct = eth_account.Account.from_key(...)
+signed = acct.sign_message(eth_account.messages.encode_defunct(sign_bytes))
+return eth_utils.to_hex(signed.signature)
+```
+
+## Register Orchestrator With Cronos Validator
+
+At last, send the orchestrator's ethereum address, cronos address, and the signature we just signed above to a Cronos
+validator, the validator should send a `set-delegate-keys` transaction to cronos network to register the binding:
+
+```shell
+$ cronosd tx gravity set-delegate-keys $val_address $orchestrator_cronos_address $orchestrator_eth_address $signature
+```
+
+## Deploy Gravity Contract On Ethereum
+
+The gravity contract can only be deployed after majority validators (66% voting powers) have registered the
+orchestrator. And before deploy gravity contract, we need to prepare the [parameters for the
+constructor](https://github.com/PeggyJV/gravity-bridge/blob/cfd55296dfb981dd7a18cefa2da9e21410fa0403/solidity/contracts/Gravity.sol#L561)
+first:
+
+- `gravity_id`. Run command `cronosd q gravity params | jq ".params.gravity_id"`
+- `threshold`, constant `2834678415`, which is just `int(2 ** 32 * 0.66)`.
+- `eth_addresses` and `powers`:
+  - Query signer set by running command: `cronosd q gravity latest-signer-set-tx | jq ".signer_set.signers"`
+  - Sum up the `power` field to get `powers`
+  - Collect the `ethereum_address` field into a list to get `eth_addresses`
+
+At last, use your favorite web3 library/tool to deploy the gravity contract with the above parameters in the ethereum
+testnet, the compiled artifacts of the contract (`Gravity.json`) can be found in [gravity-bridge's
+releases](https://github.com/PeggyJV/gravity-bridge/releases).
+
+## Run Orchestrator
```

Comment:
does this also run relayer?

Reply:
Yes, it'll relay messages in both directions.